### DISCUSSION TITLE
Plot in R Studio

### DISCUSSION THREAD

**Discussion by yohanespm77** [teamalgoritma/community#730] [Q&A] [answered]
Saya punya data seperti ini, dari tanggal 01 January s/d 31 January : 
<img width="512" alt="image" src="https://github.com/teamalgoritma/community/assets/106010284/47f0e177-0d82-4f4e-8e12-5a0dc3d610f8">

Kodingan plotnya seperti ini : 
<img width="466" alt="image" src="https://github.com/teamalgoritma/community/assets/106010284/0e550c79-7f6b-420b-afc7-4bc880d7ff0d">

        
Namun di plotnya yang muncul malah dari tanggal 31 Des s/d 01 Feb 
<img width="532" alt="image" src="https://github.com/teamalgoritma/community/assets/106010284/c9e636cc-5291-436a-a9d4-3792b897132f">

Mohon bantuannya kak, terima kasih 🙏

**Comment by arkanayudhistira** [comment]
Dear @yohanespm77,

Jika kita lihat dari hasil plot sebenarnya titik yang diplot sudah tepat. Datanya baru muncul ketika 01 Jan dan berakhir di 31 Jan, namun untuk menghilangkan tanggal tambahan mungkin dapat dicoba untuk menambahkan parameter berikut pada `scale_x_date()`:
`limits = as.Date(c('2023-01-01','2023-01-31'))` untuk membatasi tanggal.

Semoga membantu.

**Reply by yohanespm77** [reply]
Dear kak @arkanayudhistira 

Jadinya malah seperti ini kak : 
<img width="487" alt="image" src="https://github.com/teamalgoritma/community/assets/106010284/a7d4b6e8-e300-43e2-8af2-ec62aa87383f">

<img width="551" alt="image" src="https://github.com/teamalgoritma/community/assets/106010284/f8449d7c-be8f-4e9f-b3fa-a194d896a290">

**Reply by arkanayudhistira** [reply]
Untuk parameter `limits` nya dimasukkan ke dalam fungsi `scale_x_date` yang pertama, sejajar dengan parameter `date_breaks` dan `date_labels`.

**Reply by yohanespm77** [reply]
Hasilnya sama saja kak @arkanayudhistira  seperti ini : 
<img width="463" alt="image" src="https://github.com/teamalgoritma/community/assets/106010284/154ee1ea-3ae7-4035-abc4-a2344f6bb2d2">


<img width="525" alt="image" src="https://github.com/teamalgoritma/community/assets/106010284/7e1e975c-48d8-4cb2-960d-1a98fa4bdfec">

**Reply by kevbow** [reply]
Dear @yohanespm77,

Mungkin bisa dicoba ganti untuk limitnya bukan menggunakan `scale_x_date` tetapi menggunakan `xlim` seperti contoh berikut:
```r
+ xlim(as.Date(c('2023-01-01', '2023-01-31'), format="%d %b") )
```

**Comment by yohanespm77** [comment]
Tampilannya jadi seperti ini kak @kevbow

<img width="494" alt="image" src="https://github.com/teamalgoritma/community/assets/106010284/a82818c8-ddf2-4adc-b6ed-365a3211e8e8">

<img width="535" alt="image" src="https://github.com/teamalgoritma/community/assets/106010284/4947f375-c3fc-4413-ab03-9f6d20d14d45">


Bagaimana caranya agar sumbu x nya tetap berselang 1 hari ?

**Reply by kevbow** [reply]
Silakan ditambahkan parameter breaks di dalam `scale_x_date()`.
Referensi: https://ggplot2.tidyverse.org/reference/scale_date.html

**Reply by yohanespm77** [reply]
Pakai scale_x_date() malah balik lagi seperti awal plotnya kak @kevbow 

<img width="461" alt="image" src="https://github.com/teamalgoritma/community/assets/106010284/94be2034-cca5-400c-8c7d-7f8b58a4804d">


<img width="536" alt="image" src="https://github.com/teamalgoritma/community/assets/106010284/9315cb00-7e4a-4751-a105-146de6ec7aae">


Kalau scale_x_date() nya di atas xlim, tampilannya malah seperti tanpa menggunakan scale_x_date()

**Reply by divakartika** [reply, marked as answer]
Dear @yohanespm77 ,

Hal tersebut tidak dapat dilakukan dengan tipe data Datetime, hanya bisa dilakukan jika x axis bertipe data character. Berikut kami berikan contoh code-nya.

```r
dummy <- data.frame(date = as.Date(c("2023-01-01","2023-01-31")),
                    value = c(0, 0))
dummy <- dummy %>% 
  pad(start_val = ymd("2023-01-01"), end_val = ymd("2023-01-31"), interval = "1 day") %>% 
  mutate(value = runif(31, min=800, max=2000),
         date = as.character(date))

ggplot(data = dummy, mapping = aes(x = date, y = value)) +
  geom_line(group = 1) +
  geom_point() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
```

Bila informasi ini dapat menjawab pertanyaan Anda, silakan “Mark as answer” jawaban ini. Terima kasih!

![yohanes](https://github.com/teamalgoritma/community/assets/79572421/193dcf6a-91c9-4ecb-9a7c-7d776e0b694e)

**Reply by yohanespm77** [reply]
Baik, terima kasih atas informasinya kak @divakartika
